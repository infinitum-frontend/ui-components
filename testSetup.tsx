import { expect, beforeAll } from 'vitest'
import { render, RenderOptions } from '@testing-library/react'
// import '@testing-library/jest-dom/vitest' for vitest 1.x.x
import { ReactElement, JSXElementConstructor } from 'react'
import matchers from '@testing-library/jest-dom/matchers'

// extends Vitest's expect method with methods from react-testing-library
// https://github.com/testing-library/jest-dom#custom-matchers
expect.extend(matchers)

export const renderComponent = function (
  Component: ReactElement,
  options?: RenderOptions
): {
  el: HTMLElement
  container: HTMLElement
  rerender: (ui: ReactElement<any, string | JSXElementConstructor<any>>) => void
} {
  const { container, rerender } = render(Component, options)

  const el = container.firstChild as HTMLElement

  return { el, container, rerender }
}

beforeAll(() => {
  global.ResizeObserver = class ResizeObserver {
    observe(): any {}
    unobserve(): any {}
    disconnect(): any {}
  }

  // https://github.com/testing-library/react-testing-library/issues/95#issuecomment-392879866
  // https://github.com/vuejs/vue-test-utils/issues/319
  if (!HTMLElement.prototype.scrollTo) {
    HTMLElement.prototype.scrollTo = () => {}
  }
})

// runs a cleanup after each test case (e.g. clearing jsdom)
// afterEach(() => {
//   cleanup()
// })
