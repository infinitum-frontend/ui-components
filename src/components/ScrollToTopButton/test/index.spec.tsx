import { it, describe } from 'vitest'
import { renderComponent } from '../../../../testSetup'
import { ScrollToTopButton } from '../index'
import { screen } from '@testing-library/dom'

describe('ScrollToTopButton', () => {
  it('should match snapshot', () => {
    const { el } = renderComponent(<ScrollToTopButton />)
    expect(el).toMatchSnapshot()
  })

  it('should not be visible with visibilityThreshold > scrollMarginTop', () => {
    renderComponent(
      <ScrollToTopButton visibilityThreshold={500} scrollMarginTop={100} />
    )
    const scrollToTopButton = screen.queryByRole('button') as HTMLElement
    expect(scrollToTopButton).toBeInTheDocument()
    expect(scrollToTopButton).toHaveStyle({
      opacity: 0,
      'pointer-events': 'none'
    })
  })

  it('should be visible with visibilityThreshold < scrollMarginTop', () => {
    renderComponent(
      <ScrollToTopButton visibilityThreshold={0} scrollMarginTop={100} />
    )
    const scrollToTopButton = screen.queryByRole('button') as HTMLElement
    expect(scrollToTopButton).toBeInTheDocument()
    expect(scrollToTopButton).not.toHaveStyle({
      opacity: 0,
      'pointer-events': 'none'
    })
  })
})
