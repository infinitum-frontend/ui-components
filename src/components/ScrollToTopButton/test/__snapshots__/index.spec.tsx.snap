// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScrollToTopButton > should match snapshot 1`] = `
<button
  class="inf-button inf-scroll-to-top-button inf-scroll-to-top-button--hidden inf-button--variant-ghost inf-button--size-medium inf-button--square"
  type="button"
>
  <span
    class="inf-button__content"
  >
    <span
      class="inf-button__icon"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4L12 20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M16.9998 8.99996C16.9998 8.99996 13.3173 4.00001 11.9997 4C10.6821 3.99999 6.99976 9 6.99976 9"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </span>
</button>
`;
